File: codes/ruby/encapsulation/student.rb
require_relative "person"

class Student < Person

    def initialize(name)
        super(name)
        @t = "student"
        @luckyNumber = 1
        @name = "#@name 2"
        @num = 5
    end

    def info_student
        puts "info student"
        info1
        info2
    end

end

File: codes/javascript/encapsulation/student.js
var Person = require("./person.js");

module.exports = Student;

function Student(name) {
    Person.call(this, name);
    this._name = name + "2";//can be accessed
    //console.log("Student => num = ", num, ", this.num = ", this.num); //error
    luckyNumber = 1;
}

Student.prototype = Object.create(Person.prototype);
Student.constructor = Student;
